### PR TITLE
Fix Modifying a dictionary menu item

### PR DESCRIPTION
**Pull request by mherman22**
# Pull Requests Requirements

- [x] The PR title includes a brief description of the work done, including the Issue number if applicable.
- [x] The PR includes a video showing the changes for the work done.
- [x] The PR title follows conventional commit label standards.
- [x] The changes confirm to the OpenElis Global x3 Styleguide and design documentation.
- [x] The changes include tests or are validated by existing tests.
- [x] I have read and agree to the Contributing Guidelines of this project.

## Screenshots

[Add relevant screenshots here if applicable]

## Related Issue

[Add a link to the related issue or mention it here if applicable]

## Other

[Add any additional information or notes here]

### CODE REVIEW

File: frontend/src/components/admin/menu/DictionaryManagement.js [hunk 1]
```diff
@@ -272,38 +272,63 @@ function DictionaryManagement() {
...
+      console.log("contents of selected item: " + JSON.stringify(selectedItem));
       if (selectedItem) {
         setDictionaryNumber(selectedItem.id);
-        setCategory(selectedItem.categoryName);
+        setCategory(selectedItem.category);
         setDictionaryEntry(selectedItem.dictEntry);
         setLocalAbbreviation(selectedItem.localAbbreviation);
         setIsActive(yesOrNo.find((item) => item.id === selectedItem.isActive));
-        setLastUpdated(selectedItem.lastupdated); // Set the lastupdated state
+        setLastUpdated(selectedItem.lastupdated);
```
Comment (mherman22) [inline frontend/src/components/admin/menu/DictionaryManagement.js:319]: i was able to add the lastupdated field but i late realized the old ui actually doesn't send it in the payload when updating an item but rather my issue is we need to populated the `dirtyfields` attribute with changed fields as part of the payload. see you monday 😄

File: frontend/src/components/admin/menu/DictionaryManagement.js [hunk 2]
```diff
@@ -117,7 +117,31 @@ function DictionaryManagement() {
...
+            id: item.id,
+            dictEntry: item.dictEntry,
+            localAbbreviation: item.localAbbreviation,
+            isActive: item.isActive,
+            categoryName: item.dictionaryCategory
+              ? item.dictionaryCategory.categoryName
+              : "not available",
+            lastupdated: item.lastupdated,
+          }));
+          console.log("new menu list: " + JSON.stringify(menuList));
```
Comment (sherrif10) [inline frontend/src/components/admin/menu/DictionaryManagement.js:141]: Let me hope you are removing this line. A screen shot would also suffice 👍 Great work done man.

File: frontend/src/components/admin/menu/DictionaryManagement.js [hunk 3]
```diff
@@ -209,30 +208,36 @@ function DictionaryManagement() {
...
+      dirtyFieldsRef.current.add("isActive");
+    }
+
+    if (!componentMounted.current[localAbbreviation]) {
+      dirtyFieldsRef.current.add("localAbbreviation");
+    }
+
+    const dirtyFields =
+      dirtyFieldsRef.current.size > 0
+        ? `;${[...dirtyFieldsRef.current].join(";")}`
```
Comment (mherman22) [inline frontend/src/components/admin/menu/DictionaryManagement.js:225]: here we join all the contents of the dirtyFieldsRef using a `;` which is required by the servverside. so the update json to be sent would look something like:

```
{
  "id": "8",
  "selectedDictionaryCategoryId": "3",
  "dictEntry": "gram + rod378",
  "localAbbreviation": "CGV",
  "isActive": "Y",
  "lastupdated": 1717826409272,
  "dirtyFormFields": ";dictEntry;isActive;localAbbreviation"
}
```

File: src/main/java/org/openelisglobal/dictionary/controller/rest/DictionaryRestController.java [hunk 4]
```diff
@@ -145,9 +145,20 @@ public ResponseEntity<?> showNextPreviousDictionary(HttpServletRequest request)
...
+     * @param redirectAttributes
+     * @return
+     * @throws IllegalAccessException
+     * @throws InvocationTargetException
+     * @throws NoSuchMethodException
+     */
     @RequestMapping(value = "/Dictionary", method = RequestMethod.POST)
     public ResponseEntity<?> showUpdateDictionary(HttpServletRequest request,
-                                             @ModelAttribute("form") @Valid DictionaryForm form, BindingResult result, SessionStatus status,
+                                                  @RequestBody @Valid DictionaryForm form, BindingResult result, SessionStatus status,
```
Comment (mherman22) [inline src/main/java/org/openelisglobal/dictionary/controller/rest/DictionaryRestController.java:161]: the issue here was lack of @RequestBody where @RequestBody is used to indicate that a method parameter should be bound to the body of the HTTP request. It tells Spring to `deserialize` the incoming request payload and convert it into a Java object....etc!

**Pull request merged**